packages/utils/telemetry-utils/src/test/utils.spec.ts

```diff
@@ -69,3 +70,55 @@ describe("tagData", () => {
 		assert.deepEqual(taggedData, expected);
 	});
 });
+
+describe("Sampling", () => {
+	let events: ITelemetryBaseEvent[] = [];
+	function getBaseLoggerWithConfig(
+		configDictionary?: Record<string, ConfigTypes>,
+	): ITelemetryBaseLogger {
+		const logger: ITelemetryBaseLogger = {
+			send(event: ITelemetryBaseEvent): void {
+				events.push(event);
+			},
+		};
+		const configProvider = (settings: Record<string, ConfigTypes>): IConfigProviderBase => ({
+			getRawConfig: (name: string): ConfigTypes => settings[name],
+		});
+		return mixinMonitoringContext(logger, configProvider(configDictionary ?? {})).logger;
+	}
+
+	beforeEach(() => {
+		events = [];
+	});
+
+	it("Systematic Sampling works as expected", () => {
```

Comment:
can you add some tests that are more like real usage? usually there is data that is tracked, and part of the event, there really isn't a reason to sample a constant event

Reply:
i think you'll find it would be convenient to support tracking some data in the functions, otherwise consumer need to track multiple pieces themselves.

Reply:
Can you expand on what you mean by this and are there any examples of this in our code base? I added a more complex test example, perhaps that encompasses what you envisioned.

Reply:
the best way to illustrate is probably to move some code somewhere to the new pattern.